opensearch-operator/pkg/helpers/helpers.go

```diff
@@ -359,3 +360,28 @@ func CompareVersions(v1 string, v2 string) bool {
 	ver2, _ := version.NewVersion(v2)
 	return err == nil && ver1.LessThan(ver2)
 }
+
+func GetJvmHeapSize(nodePool *opsterv1.NodePool) string {
+	jvmHeapSizeTemplate := "-Xmx%s -Xms%s"
+
+	if nodePool.Jvm == "" {
+		memoryLimit := nodePool.Resources.Requests.Memory()
+		nodePoolMemorySize, _ := resource.ParseQuantity(memoryLimit.String())
+
+		// Memory request is not present
+		if nodePoolMemorySize.IsZero() {
```

Comment:
You must check for the error here, zero memorysze is different from the quantity not being parsed even if you treat them the same.
```suggestion
		if err != nil || nodePoolMemorySize.IsZero() {
```

Reply:
This does not need it at the moment because I'm using nodePool.Resources.Requests.Memory() which is a quantity already to make all the calculations without creating additional quantities.